pkg/skaffold/logfile/logfile.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright 2020 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logfile
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+)
+
+// Create creates or truncates a file to be used to output logs.
+func Create(name string) (*os.File, error) {
+	root := filepath.Join(os.TempDir(), "skaffold")
+	if err := os.MkdirAll(root, 0700); err != nil {
+		return nil, fmt.Errorf("unable to temp directory %q: %w", root, err)
+	}
+
+	path := filepath.Join(root, escape(name))
+	return os.OpenFile(path, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0600)
```

Comment:
wouldn't this also imply that successive runs of `skaffold build` would append logs to the same file if the image names matched up? i.e. if caching was disabled

Reply:
This code will truncate the log file on each `Create()`. The caller can avoid calling `Create()` with the same name for now.